pennylane/ops/qubit/arithmetic_ops.py

```diff
@@ -118,12 +118,34 @@ def _matrix(cls, *params):
         return QubitCarry._mat
 
     @staticmethod
-    def decomposition(wires):
-        decomp_ops = [
+    def compute_decomposition(wires):
+        """Determine ``QubitCarry``'s decomposition for specified parameters, wires,
+        and hyperparameters. The decomposition defines an Operator as a product of
+        more fundamental gates:
+
+        .. math:: O = O_1 O_2 \dots O_n.
+
+        ``compute_decomposition`` is a static method and can provide the decomposition of a given
+        operator without creating a specific instance. The instance method ``decomposition`` uses
+        this method and the instance's variables.
```

Comment:
In the other PRs I kept the specific docstring short, and only added explanations like this paragraph to the top-level docstring. 

But I'm also not sure what's the best idea here.